File: adb.ts
const totalSteps = (staircase: number[], stepsPerFlight: number) =>
    staircase.reduce((acc, curr) => acc + Math.ceil(curr/stepsPerFlight), 0)
    + (staircase.length - 1) * 2

console.log(totalSteps([15], 2));
console.log(totalSteps([15, 15], 2));
console.log(totalSteps([5, 11, 9, 13, 8, 30, 14], 3));

File: src/main/scala/adb.scala
object adb {

  def totalSteps(stepsPerFlight: Int)(staircase: Array[Int]): Int = {
    assert(staircase.length >= 1 && staircase.length <= 50, "The staircase is not conform.")
    assert(staircase.forall(staircase => staircase >= 5 && staircase <= 30), "Not all flights of stairs are conform.")
    assert(stepsPerFlight >= 2 && stepsPerFlight <= 5, "Steps per flight is not conform.")

    staircase.foldLeft(0)((acc, curr) => acc + Math.ceil(curr/stepsPerFlight.toDouble).toInt) + (staircase.length - 1) * 2
  }
}
